src/shrpx.cc

```diff
@@ -1088,6 +1111,13 @@ int call_daemon() {
 #ifdef __sgi
   return _daemonize(0, 0, 0, 0);
 #else  // !__sgi
+#ifdef HAVE_LIBSYSTEMD
+  if (sd_booted()) {
+    LOG(NOTICE) << "Daemonising disabled under systemd";
+    chdir("/");
+    return 0;
+  }
```

Comment:
This actually removes the opportunity from users to run nghttpx in daemon mode without systemd.

Perhaps, it might be nicer to have new option like --systemd option, and only do systemd related stuff, including chdir("/") and sending message to systemd with this option turned on.  I'm not totally sure that sd_notifyf is really noop if nghttpx is not started by systemd.